App.js

```diff
@@ -1,38 +1,54 @@
 import React, { Component } from 'react';
-import { Platform, StyleSheet, Text, View } from 'react-native';
-import { Constants, Location, Permissions } from 'expo';
+import { StyleSheet, Text, View } from 'react-native';
+import { Constants, Permissions } from 'expo';
+import { db } from './firebase';
 
 export default class App extends Component {
   state = {
+    trainId: 1,
     location: null,
     errorMessage: null,
   };
 
   componentWillMount() {
-    this._getLocationAsync();
+    this._requestLocationPermission();
   }
 
   componentDidMount() {
-    Location.watchPositionAsync(
-      {
-        enableHighAccuracy: true,
-      },
-      location => {
-        this.setState(location)
-      }
-    );
+    this._updateLocation();
+  }
+
+  componentWillUnmount() {
+    navigator.geolocation.clearWatch();
   }
 
-  _getLocationAsync = async () => {
-    let { status } = await Permissions.askAsync(Permissions.LOCATION);
+  _requestLocationPermission = async () => {
+    const { status } = await Permissions.askAsync(Permissions.LOCATION);
     if (status !== 'granted') {
       this.setState({
         errorMessage: 'Permission to access location was denied',
       });
     }
+  };
 
-    let location = await Location.getCurrentPositionAsync({});
-    this.setState({ location });
+  _updateLocation = () => {
```

Comment:
I think having the _ is little not neat as it not provides any private or public methods.

Reply:
The naming for the `_updateLocation` is unclear. It is doing a lot of things and not directly based from the method definition.

Suggestions: 
- Create a functional method that will transmit the location
- Create a method that will trigger the position watching